provisioner/aws-manager.js

```diff
@@ -0,0 +1,696 @@
+    var rState = state[region] = {};
+
+    function x(type) {
+      if (!rState[type]) {
+        rState[type] = {
+          running: [],
+          pending: [],
+          spotReq: [],
+        };
+      }
+    }
+
+    instanceState[region].Reservations.forEach(function(reservation) {
+      reservation.Instances.forEach(function(instance) {
+        var workerType = instance.KeyName.substr(that.keyPrefix.length);
+        x(workerType);
+        rState[workerType][instance.State.Name].push(instance); 
+      });
+    });
+
+    spotReqs[region].SpotInstanceRequests.forEach(function(request) {
+      var workerType = request.LaunchSpecification.KeyName.substr(that.keyPrefix.length);
+    }); 
+
+  });
+
+  return state;
+};
+
+
+/**
+ * Get Api State Only
+ */
+AwsManager.prototype.getApi = function(region, type) {
+  if (region && type) {
+    if (!this.__apiState[region]) {
+      return {};
+    }
+    return this.__apiState[region][type];
+  } else if (region && !type) {
+    return this.__apiState[region];
+  } else if (!region && !type) {
+    return this.__apiState;
+  }
+};
+
+/**
+ * Get Internal State Only.  Remember that Internal state
+ * contains an extra Container object which contains
+ * the the raw `request`, the `workerType` name, the `bid`
+ * and a datetime of when it was `submitted`
+ */
+AwsManager.prototype.getInternal = function(region, type) {
+  if (region && type) {
+    if (!this.__internalState[region]) {
+      return {};
+    }
+    return this.__internalState[region][type];
+  } else if (region && !type) {
+    return this.__internalState[region];
+  } else if (!region && !type) {
+    return this.__internalState;
+  }
+};
+
+
+/**
+ * List the types known in a given region
+ */
+AwsManager.prototype.typesForRegion = function(region) {
+  assert(region);
+  var apiState = this.getApi(region) || {};
+  var internalState = this.getInternal(region) || {};
+
+  var types = Object.keys(apiState);
+  Array.prototype.push.apply(types, Object.keys(internalState).filter(function(type) {
+    return !types.includes(type);
+  }));
+  return Object.keys(this.__apiState[region]);
+};
+
+
+/**
+ * List the regions that this Manager is configured to 
+ * manage
+ */
+AwsManager.prototype.managedRegions = function () {
+  return lodash.clone(this.ec2.regions);
+};
+
+/**
+ * Return a list of workerTypes known to AWS
+ */
+AwsManager.prototype.knownWorkerTypes = function() {
+  var workerTypes = [];
+  var that = this;
+
+  this.managedRegions().forEach(function(region) {
+    that.typesForRegion(region).forEach(function(workerType) {
+      if (workerTypes.includes(workerType)) {
+        workerTypes.push(workerType);
+      }
+    });
+  });
+
+  return workerTypes;
+};
+
+
+/**
+ * Return a list of all running Instance Ids that are known in this AWS State
+ * These are not categorized by region.  It's one list of strings.
+ */
+AwsManager.prototype.listRunningInstanceIds = function() {
```

Comment:
Nvm, it's clear what these unused functions do...
And I can see that using `listSpotUnfulfilledRequestIds` to check for old spot requests to cancel because we never expect them to finish is a good idea.